custom_model_runner/datarobot_drum/drum/gpu_predictors/vllm_predictor.py

```diff
@@ -0,0 +1,96 @@
+"""
+Copyright 2024 DataRobot, Inc. and its affiliates.
+All rights reserved.
+This is proprietary source code of DataRobot, Inc. and its affiliates.
+Released under the terms of DataRobot Tool and Utility Agreement.
+"""
+import os
+import signal
+import subprocess
+import typing
+
+import requests
+from requests import Timeout
+
+from datarobot_drum.drum.enum import CustomHooks
+from datarobot_drum.drum.exceptions import DrumCommonException
+from datarobot_drum.drum.gpu_predictors import BaseGpuPredictor
+from datarobot_drum.resource.drum_server_utils import DrumServerProcess
+
+
+class VllmPredictor(BaseGpuPredictor):
+    def __init__(self):
+        super().__init__()
+
+    def health_check(self) -> typing.Tuple[dict, int]:
+        """
+        Proxy health checks to NeMo Inference Server
+        """
+        try:
+            health_url = f"{self.openai_host}:{self.health_port}/v1/health/ready"
+            response = requests.get(health_url, timeout=5)
+            return {"message": response.text}, response.status_code
+        except Timeout:
+            return {"message": "Timeout waiting for Nemo health route to respond."}, 503
+
+    def download_and_serve_model(self, openai_process: DrumServerProcess):
+        if self.python_model_adapter.has_custom_hook(CustomHooks.LOAD_MODEL):
+            try:
+                self.python_model_adapter.load_model_from_artifact(skip_predictor_lookup=True)
+            except Exception as e:
+                raise DrumCommonException(f"An error occurred when loading your artifact: {str(e)}")
+
+        cmd = [
+            "nemollm_inference_ms",
+            "--model_name",
+            self.model_name,
+            "--health_port",
+            self.health_port,
+            "--openai_port",
+            self.openai_port,
+            "--nemo_port",
+            self.nemo_port,
+            "--num_gpus",
+            self.gpu_count,
+            "--log_level",
+            "info",
+        ]
```

Comment:
it should be a different command for vLLM, e.g.
```
        cmd = [
            "python",
            "-m",
            "vllm.entrypoints.openai.api_server",
            "--model",
            self.model_name,
        ]
```

Reply:
ah, you're starting it in the `start_server.sh`. So it's not needed there.

Reply:
I'm not finished with this yet. I'll work on this in a follow on but I want to get the refactor in because it is disruptive.